src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -2125,6 +2125,11 @@ private void recordRelocation(IntPtr _this, void* location, void* target, ushort
                         throw new NotImplementedException("RuntimeFieldHandle is not implemented");
                     }
 
+                    if (targetObject is TypeDesc)
+                    {
+                        targetObject = _compilation.NodeFactory.NecessaryTypeSymbol((TypeDesc)targetObject);
+                    }
```

Comment:
We do not want to patch it up after the fact here. This should be fixed in embedGenericHandle instead, something like:

```
            else if (!fEmbedParent && pResolvedToken.hField != null)
            {
                pResult.handleType = CorInfoGenericHandleType.CORINFO_HANDLETYPE_FIELD;
                pResult.compileTimeHandle = (CORINFO_GENERIC_STRUCT_*)pResolvedToken.hField;

                // fRuntimeLookup = th.IsSharedByGenericInstantiations() && pFD->IsStatic();

                // TODO: shared generics
                // If the target is not shared then we've already got our result and
                // can simply do a static look up
                pResult.lookup.lookupKind.needsRuntimeLookup = false;

                pResult.lookup.constLookup.handle = (CORINFO_GENERIC_STRUCT_*)pResult.compileTimeHandle;
                pResult.lookup.constLookup.accessType = InfoAccessType.IAT_VALUE;
            }
            else
            {
                TypeDesc td = HandleToObject(pResolvedToken.hClass);

                pResult.handleType = CorInfoGenericHandleType.CORINFO_HANDLETYPE_CLASS;
                pResult.compileTimeHandle = (CORINFO_GENERIC_STRUCT_*)pResolvedToken.hClass;

                // TODO? If we're embedding a method handle for a method that points to a sub-class of the actual
                //       class, we might need to embed the actual declaring type in compileTimeHandle.  

                // TODO: shared generics
                // IsSharedByGenericInstantiations would not work here. The runtime lookup is required
                // even for standalone generic variables that show up as __Canon here.
                //fRuntimeLookup = th.IsCanonicalSubtype();

                // If the target is not shared then we've already got our result and
                // can simply do a static look up
                pResult.lookup.lookupKind.needsRuntimeLookup = false;

                pResult.lookup.constLookup.handle = (void*)ObjectToHandle(_compilation.NodeFactory.NecessaryTypeSymbol(td),
                pResult.lookup.constLookup.accessType = InfoAccessType.IAT_VALUE;
            }
```